### PR TITLE
fix(dedicated.account): refactor and add siret module

### PR DESCRIPTION
**Pull request by ghyenne**
| Question         | Answer
| ---------------- | ---
| Branch?          | `fix/MANAGER-8600` 
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | MANAGER-9844 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
I refactored the account modification section and added siret module for corporation FR

### 🎏  Translations

- [x] TRANS-48492

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.controller.js [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+
+    // handle special phone prefix case
+    if (this.getFieldType() === 'phone') {
+      this.$scope.$watch(
+        '$ctrl.newAccountForm.rules',
+        (rules) => {
+          const rule = rules.find(
+            (value) => value.fieldName === 'phoneCountry',
+          );
+          this.phoneCountryList = rule.in.map((country) => ({
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.controller.js:65]:
```suggestion
          this.phoneCountryList = rule?.in.map((country) => ({
```

File: packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.html [hunk 2]
```diff
@@ -122,19 +135,9 @@
...
-        ></p>
-    </div>
 
-    <!-- PASSWORD INPUT -->
-    <div data-ng-if="$ctrl.getFieldType() === 'password'">
-        <input
-            type="password"
-            class="form-control"
+        <oui-password
+            data-ng-if="$ctrl.getFieldType() === 'password'"
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.html:140]: prefer moving 'password' to constant.

File: packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.controller.js [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+      });
+    }
+
+    // handle special phone prefix case
+    if (this.getFieldType() === 'phone') {
+      this.$scope.$watch(
+        '$ctrl.newAccountForm.rules',
+        (rules) => {
+          const rule = rules.find(
+            (value) => value.fieldName === 'phoneCountry',
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.controller.js:63]: Prefer moving "phoneCountry" to constant file

File: packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.controller.js [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+        }
+
+        if (this.rule.regularExpression) {
+          if (this.rule.prefix && value) {
+            value = this.rule.prefix + value;
+          }
+          const regex = this.rule.regularExpression.replace(/\//g, '\\/');
+          return new RegExp(regex).test(value);
+        }
+        if (this.rule.fieldName === 'password') {
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.controller.js:136]: same as https://github.com/ovh/manager/pull/8312/files#r1009210689

File: packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/new-account-form-controller.js [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+        this.initError = err.data.message || err.message || err;
+      })
+      .finally(() => {
+        this.loading = false;
+      });
+  }
+
+  // initialize rules with /me data
+  initializeRulesWithOriginalModel(rules) {
+    forEach(this.originalModel, (value, key) => {
```
Comment (sachinrameshn) [inline packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/new-account-form-controller.js:63]: prefer: this.originalModel.forEach()

File: packages/manager/apps/dedicated/client/app/account/user/infos/user-infos.service.js [hunk 6]
```diff
@@ -168,31 +167,28 @@ export default /* @ngInject */ function UserAccountInfosService(
...
-        if (models) {
-          const rules = models['nichandle.CreationRules'];
-          if (rules) {
-            result = keys(rules.properties);
-          }
-        }
-
-        return result;
+      .then(({ data }) => {
+        const { models } = data;
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/account/user/infos/user-infos.service.js:178]:
```suggestion
        const { models = {} } = data;
```

File: packages/manager/apps/dedicated/client/app/account/user/infos/user-infos.service.js [hunk 7]
```diff
@@ -168,31 +167,28 @@ export default /* @ngInject */ function UserAccountInfosService(
...
+        return Object.keys(models['nichandle.CreationRules'].properties) || [];
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/account/user/infos/user-infos.service.js:179]:
```suggestion
        return Object.keys(models['nichandle.CreationRules']?.properties) || [];
```

File: packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.controller.js [hunk 8]
```diff
@@ -0,0 +1,396 @@
...
+    this.$translate = $translate;
+    this.atInternet = atInternet;
+  }
+
+  $onInit() {
+    // unique field identifier
+    this.id = `ovh_field_${this.rule.fieldName}`;
+
+    // field value
+    this.value = undefined;
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.controller.js:30]: I don't have the same result for checkbox replace by null;

File: packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.controller.js (hunk 1 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.controller.js]: It's applied

File: packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.html (hunk 2 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.html]: It's applied

File: packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.controller.js (hunk 4 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.controller.js]: It's applied

File: packages/manager/apps/dedicated/client/app/account/user/infos/user-infos.service.js (hunk 6 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/account/user/infos/user-infos.service.js]: It's applied

File: packages/manager/apps/dedicated/client/app/account/user/infos/user-infos.service.js (hunk 7 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/account/user/infos/user-infos.service.js]: It's applied

**Review by sachinrameshn — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.html [hunk 9]
```diff
@@ -49,69 +35,96 @@
...
+            data-ng-if="$ctrl.getFieldType() === $ctrl.FIELD_NAME_LIST.phone"
         >
-            <span
-                class="input-group-addon"
-                data-ng-if="$ctrl.rule.prefix"
-                data-ng-bind="$ctrl.rule.prefix"
+            <ui-select
+                data-ng-model="$ctrl.phoneCountry.country"
+                data-on-select="$ctrl.onPhonePrefixChange()"
+                theme="ui-select-oui-kit"
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.html:90]:
```suggestion
                data-theme="ui-select-oui-kit"
```

File: packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.html [hunk 10]
```diff
@@ -144,100 +147,43 @@
...
-            data-ng-options="val as val.translated for val in $ctrl.getTranslatedEnums() track by val.key"
+        <!-- CHECKBOX INPUT -->
+        <oui-checkbox
+            data-ng-if="$ctrl.getFieldType() === 'checkbox'"
             data-ng-attr-id="{{ 'ovh_field_' + $ctrl.rule.fieldName }}"
             data-ng-attr-name="{{ 'ovh_field_' + $ctrl.rule.fieldName }}"
-            data-ng-model="$ctrl.value"
-            data-ng-change="$ctrl.onChange()"
+            model="$ctrl.value"
+            on-change="$ctrl.onChange()"
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.html:159]:
```suggestion
            data-model="$ctrl.value"
            data-on-change="$ctrl.onChange()"
```

**Review by anooparveti — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.html (hunk 9 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.html]: It's applied

File: packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.html (hunk 10 above)
Comment (ghyenne) [reply packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.html]: It's applied

**Pull request merged**